### PR TITLE
fix(hosted-hpb): Correctly handle API response when the account expir…

### LINKED ISSUE #9656: Error "Getting the account information failed: account not found" when HPB trial expired
### ⚠️ This issue respects the following points: ⚠️

- [X] This is a **bug**, not a question or a configuration/webserver/proxy issue.
- [X] This issue is **not** already reported on Github _(I've searched it)_.
- [X] Nextcloud Server **is** up to date. See [Maintenance and Release Schedule](https://github.com/nextcloud/server/wiki/Maintenance-and-Release-Schedule) for supported versions.
- [X] Nextcloud Server **is** running on 64bit capable CPU, PHP and OS.
- [X] I agree to follow Nextcloud's [Code of Conduct](https://nextcloud.com/contribute/code-of-conduct/).

### Bug description

The logfile is generating these entries from Nextcloud-App **Spreed**:

`{"reqId":"xrWYwfDNdMZX0GBjIupY","level":3,"time":"2023-05-28T15:30:06+00:00","remoteAddr":"","user":"--","app":"spreed","method":"","url":"--","message":"**Getting the account information failed: account not found - HTTP status: 404** Response body: ","userAgent":"--","version":"26.0.2.1","data":{"app":"spreed"},"id":"6473762837795"}`

### Steps to reproduce

1. No speacial activity
2. Check NC Logfiles
3. Showing errors by App Spreed

### Expected behavior

No log errors

### Installation method

Community Manual installation with Archive

### Nextcloud Server version

26

### Operating system

Debian/Ubuntu

### PHP engine version

PHP 8.0

### Web server

Apache (supported)

### Database engine version

MariaDB

### Is this bug present after an update or on a fresh install?

None

### Are you using the Nextcloud Server Encryption module?

Encryption is Disabled

### What user-backends are you using?

- [ ] Default user-backend _(database)_
- [ ] LDAP/ Active Directory
- [ ] SSO - SAML
- [X] Other

### Configuration report

```shell
This version of Nextcloud requires at least PHP 8.0<br/>You are currently running 7.3.3. Please update your

This seems strange as my PHP is 8 and Nextcloud shows the correct configuration.
```

### List of activated Apps

```shell
This version of Nextcloud requires at least PHP 8.0<br/>You are currently running 7.3.3. Please update your

This seems strange as my PHP is 8 and Nextcloud shows the correct configuration.
```

### Nextcloud Signing status

```shell
No errors have been found.
```

### Nextcloud Logs

```shell
{"reqId":"xrWYwfDNdMZX0GBjIupY","level":3,"time":"2023-05-28T15:30:06+00:00","remoteAddr":"","user":"--","app":"spreed","method":"","url":"--","message":"Getting the account information failed: account not found - HTTP status: 404 Response body: ","userAgent":"--","version":"26.0.2.1","data":{"app":"spreed"}}
```

### Additional info

_No response_

### PR DESCRIPTION
**Pull request by nickvergessen**
…ed or was deleted

### ☑️ Resolves

* Fix #9656 

## 🛠️ API Checklist

- Fixed an issue when the HPB expired (401 can be returned together with an empty response)
- Improved the notifications (and fixed missing translations)

Before | After
---|---
grafik | Bildschirmfoto vom 2023-11-22 12-24-54

### 🏁 Checklist

- [x] ⛑️ Tests (unit and/or integration) are included or not possible
- [x] 📘 API documentation in `docs/` has been updated or is not required
- [x] 🔖 Capability is added or not needed

### CODE REVIEW

File: lib/BackgroundJob/CheckHostedSignalingServer.php [hunk 1]
```diff
@@ -67,9 +68,17 @@ protected function run($argument): void {
...
 			$accountInfo = $this->hostedSignalingServerService->fetchAccountInfo($accountId);
-		} catch (HostedSignalingServerAPIException $e) { // API or connection issues
-			// do nothing and just try again later
-			return;
+		} catch (HostedSignalingServerAPIException $e) {
+			if ($e->getCode() === Http::STATUS_NOT_FOUND) {
+				// Account was deleted, so remove the information locally
+				$accountInfo = ['status' => 'deleted'];
+			} elseif ($e->getCode() === Http::STATUS_UNAUTHORIZED) {
+				// Account is expired and deletion is pending unless it's reactivated.
```
Comment (nickvergessen) [inline lib/BackgroundJob/CheckHostedSignalingServer.php:76]: it seems previously expired accounts always received a 200 with body. Now it's a 401. Deleted accounts receive a 404.

So both cases are now specifically catched and handled so the status notifications happen as planned.

**Review by Antreesy — APPROVED** [review]
All that is close to my level of understanding, seems OK

**Pull request merged**